src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs

```diff
@@ -156,17 +157,49 @@ private Task Run(CancellationToken ct)
 
         private async Task<Stream> ConnectToServer(CancellationToken token)
         {
-            var command = new GetTcpInfoCommand("Subscription/" + _dbName, _dbName);
+            async Task<TcpConnectionInfo> LegacyTryGetTcpInfo(RequestExecutor requestExecutor1, JsonOperationContext context, ServerNode node = null)
+            {
+                var tcpCommand = new GetTcpInfoCommand("Subscription/" + _dbName, _dbName);
+                try
+                {
+                    if(node != null)
+                    {
+                        await requestExecutor1.ExecuteAsync(node, null, context, tcpCommand, shouldRetry: false, sessionInfo: null, token: token)
+                        .ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        await requestExecutor1.ExecuteAsync(tcpCommand, context, sessionInfo: null, token: token)
+                            .ConfigureAwait(false);
+                    }                    
+                }
+                catch (Exception _)
+                {
+                    _redirectNode = null;
+                    throw;
+                }
+
+                return tcpCommand.Result;
+            }
+
+            var command = new GetTcpInfoForRemoteTaskCommand("Subscription/" + _dbName, _dbName, _options?.SubscriptionName, verifyDatabase:true);
 
             var requestExecutor = _store.GetRequestExecutor(_dbName);
 
+            TcpConnectionInfo tcpInfo;
             using (requestExecutor.ContextPool.AllocateOperationContext(out JsonOperationContext context))
             {
                 if (_redirectNode != null)
                 {
                     try
                     {
-                        await requestExecutor.ExecuteAsync(_redirectNode, null, context, command, shouldRetry: false, sessionInfo: null, token: token).ConfigureAwait(false);
+                        await requestExecutor.ExecuteAsync(_redirectNode, null, context, command, shouldRetry: false, sessionInfo: null, token: token)
+                            .ConfigureAwait(false);
+                        tcpInfo = command.Result;
+                    }
+                    catch (ClientVersionMismatchException)
```

Comment:
what would throw this error?

Reply:
the router when it doesn't find the path for the endpoint, I have tested this by modifying the remote task URL to something that doesn't exist and it propagate nicely to the client